fontawesome_markdown/main.py

```diff
@@ -3,22 +3,46 @@
 from markdown.inlinepatterns import Pattern
 from markdown.util import etree
 from .icon_list import icons
+from .styleutil import prefix_to_style, style_to_prefix
+import json
 
-fontawesome_pattern = r':fa-([-\w]+):'
+fontawesome_pattern = r':(fa[bsrl]?)?\s?fa-([-\w]+)\s?(fa-(xs|sm|lg|[\d+]x|10x))?:'
 
 
 class FontAwesomeException(Exception):
-    'Exception for unknown icon names'
+    'Exception for unknown icon name, prefix or size'
     pass
 
 
 class FontAwesomePattern(Pattern):
     'Markdown pattern class for matching things that look like FA icons'
     def handleMatch(self, m):
         el = etree.Element('i')
-        icon_name = m.group(2)
+        prefix = m.group(2)
+        icon_name = m.group(3)
+        size = m.group(4)
         if icon_name in icons:
-            el.attrib = {'class': 'fa fa-{0}'.format(icon_name)}
+            styles = icons[icon_name]
+            if not prefix and 'solid' in styles:
+                # default prefix is fa
+                prefix = 'fa'
+            elif not prefix and 'solid' not in styles:
+                style = styles[0]
+                prefix = style_to_prefix(style)
```

Comment:
As mentioned in the other comment, the `style_to_prefix` function can just be a `dict` and here the `.get` method can be used:
``` python
style_to_prefix = {
    'fas': 'solid',
    'fa': 'solid',
    'fab': 'brands',
    # etc
}
prefix = style_to_refix.get(style)
```

Reply:
fixed